packages/wrangler/src/create-worker-upload-form.ts

```diff
@@ -172,10 +172,6 @@ export function createWorkerUploadForm(worker: CfWorkerInit): FormData {
         );
         // And then remove it from the modules collection
         modules = modules?.filter((m) => m !== module);
-      } else if (module.type === "buffer") {
-        throw new Error(
-          'The "buffer" module type is not yet supported in service worker format workers'
-        );
```

Comment:
Is this change specific to this PR? It kind of feels like it is an independent fix? But I might be misunderstanding its importance in terms of text_blobs?

Reply:
This just rounds out the story for module system support. With module workers, we can import modules of all the supported types: js, wasm, text, and data. With service-workers, we can import js, wasm, text, but _not_ data. We can detect its usage and throw an error. I simply moved the error to earlier in the pipeline, right when we're building the worker, instead of waiting for when we create the worker form upload. 

I don't expect this error to trigger often; there are no service-worker workers right now that import a Data module (because there's no equivalent like `wasm_modules` or `text_blobs` for data.) The usecase this will probably trigger is when a third party library imports a Data module, and is used in a service-worker (It'll work fine with a module worker). If that happens, and it's a good/common library, then the thing to do will be to convince the runtime team to add a new type of binding (`data_buffers`?). We can then leverage that to add support for Data modules for service-workers.

Reply:
I just checked internally, and we _do_ have support for a `data_blob` binding. Filed https://github.com/cloudflare/wrangler2/issues/740